### PR TITLE
Dbatiste/setup

### PR DESCRIPTION
**Pull request by dbatiste**

### CODE REVIEW

**Comment by dbatiste** [comment]
@dlockhart, @omsmith  : the purpose of this module is to provide common eslinting options that can be used as per [eslint shareable config configuration](http://eslint.org/docs/developer-guide/shareable-configs.html).  Any consumers can override with their own options as they normally would, but we can at least start with a common set of options and go from there.  I was hoping to get your thoughts before publishing.

I've copied these settings from the [react-valence-ui-fileviewer](https://github.com/Brightspace/react-valence-ui-fileviewer/blob/master/.eslintrc) eslint file, and are identical to [content-fileviewer](https://github.com/Brightspace/content-fileviewer/blob/master/.eslintrc), but different from other eslint config files for some of our modules.  These options are currently layered on-top of the eslint:recommended rules, but that can easily be changes if we want to completely ignore the recommended set.  Nevertheless, in almost all cases, I think we should be able to start with a common config, and the consumer just overrides if necessary.

I've marked the ones that are different from the eslint:recommended set.  I am not sure if the others are intentionally listed out... if not, they could be removed since we'll automatically get them from the recommended set.

Once deployed, using this is merely a matter of including a reference to it and setting up your `.eslintrc` file to extend the brightspace config with any project specific overrides.  For example, to get our default linting options:

```
{
  "extends": "brightspace"
}
``` 
Or to get linting options to roll with React (JSX, Jest, etc), you could specify:
```
{
  "extends": "brightspace/react-config",
  "parser": "babel-eslint"
}
```
And if someone wanted to override rules, they just do what they normally would:
```
{
  "extends": "brightspace/react-config",
  "parser": "babel-eslint",
  "rules": {
    "no-undef": 0
  }
}
```

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2014 D2L Corporation
```
Comment (dlockhart) [inline LICENSE]: It's 2015 now :)

Comment (dbatiste) [inline LICENSE]: Thx.  Will fix that.  :)

File: README.md [hunk 2]
```diff
@@ -1,2 +1,55 @@
...
+}
+```
+
+### React Config
+
+Specify the `react-config` config for the `extends` property: 
+
+```javascript
+{
+  "extends": "brightspace/react-config",
```
Comment (omsmith) [inline README.md]: How does.. how does this magically work? Does eslint prefix _eslint-config-_?

**Comment by dlockhart** [comment]
LGTM Dave, thanks for adding!

File: README.md (hunk 2 above)
Comment (omsmith) [inline README.md]: The link about the shareables says yes. neat!

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+charset = utf8
+end_of_line = lf
+trim_trailing_whitespace = true
+insert_final_newline = true
+indent_style = tab
+indent_size = 4
+
+[{package.json,.travis.yml}]
+indent_style = space
+indent_size = 2
```
Comment (omsmith) [inline .editorconfig]: Kinda funny that the file that says `insert_final_newline = true` as a rule, doesn't have a final newline.

**Comment by omsmith** [comment]
Sweet, wanna PR our 2 projects with this after too also as well?

File: README.md (hunk 2 above)
Comment (dbatiste) [inline README.md]: Yeah - I actually was hoping this was not a requirement, and tried to use a module name that began with 'brightspace-' similar to a couple other repos/modules we have, but unfortunately it seems I had to follow suit with 'eslint-config-'.

File: .editorconfig (hunk 3 above)
Comment (dbatiste) [inline .editorconfig]: Thx for pointing that out.  I copied that file from elsewhere.  I'll correct.  :)

**Comment by dbatiste** [comment]
Sounds good.  Will do.  :)

**Comment by dbatiste** [comment]
The react config now includes the react plugin rules, same as the modules mentioned above.  It is worth noting that the plugin only enables [one rule by default](https://github.com/yannickcr/eslint-plugin-react/blob/master/index.js), specific to correct handling unused variables that are actually used in JSX.

**Pull request merged**